File: HomeworkManager.Angular/ClientApp/src/app/shared-module/models/assignment/assignment-model.ts
import { AssignmentTypeId } from "./assignment-type-id";

export class AssignmentModel {
  assignmentId!: number;
  name!: string;
  description: string | null = null;
  deadline!: string;
  presentationRequired: boolean = false;
  isDraft: boolean = true;
  assignmentTypeId: AssignmentTypeId | null = null;
  assignmentTypeName: string | null = null;
  courseId!: number
  courseName!: string
  groupId!: number
  groupName!: string
}

File: HomeworkManager/HomeworkManager.Model/CustomEntities/Assignment/AssignmentModel.cs
using HomeworkManager.Model.Constants;

namespace HomeworkManager.Model.CustomEntities.Assignment;

public class AssignmentModel
{
    public required int AssignmentId { get; set; }
    public required string Name { get; set; }
    public required string? Description { get; set; }
    public required DateTime Deadline { get; set; }
    public required bool PresentationRequired { get; set; }
    public required bool IsDraft { get; set; }
    public required AssignmentTypeId? AssignmentTypeId { get; set; }
    public required string? AssignmentTypeName { get; set; }
    public required int CourseId { get; set; }
    public required string CourseName { get; set; }
    public required string GroupName { get; set; }
}